db_test.go

```diff
@@ -2287,3 +2287,77 @@ func TestOpenDBReadOnly(t *testing.T) {
 	require.Equal(t, 20, count)
 	require.NoError(t, db.Close())
 }
+
+func TestBannedPrefixes(t *testing.T) {
+	dir, err := ioutil.TempDir("", "badger-test")
+	require.NoError(t, err, "temp dir for badger count not be created")
+	defer os.RemoveAll(dir)
+
+	ops := getTestOptions(dir)
+	// All values go into vlog files. This is for checking if banned keys are properly decoded on DB
+	// restart.
+	ops.ValueThreshold = 0
+	ops.ValueLogMaxEntries = 2
+	db, err := Open(ops)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(db.vlog.filesMap))
+
+	var keys [10][]byte
+	var bannedPrefixes [][]byte
+	isBanned := func(key []byte) bool {
+		for _, prefix := range bannedPrefixes {
```

Comment:
Don't store the bannedPrefixes in the test. Fetch it using `db.GetBannedPrefixes`

Reply:
Added a separate check to comparision with `db.GetBannedPrefixes()`. The local list acts as the source of truth.